### DISCUSSION TITLE
How to bundle maven dependencies like with ebr

### DISCUSSION THREAD

**Discussion by git-sources** [eclipse-tycho/tycho#3232] [Q&A] [answered]
I would like to bundle multiple maven dependencies into one osgi jar.
With ebr it was possible to specify multiple dependencies and create a bundle from it.
For example I would like to bundle org.apache.logging- log4j-1.2-api, log4j-api and log4j-core.

Thanks.

**Comment by laeubi** [comment]
You maybe should explain what exactly you want to archive, do you want *one* bundle that includes all of those or do you want multiple bundles generated at once and how do you want to use them with Tycho.

**Comment by git-sources** [comment, marked as answer]
I want one bundle which contains all those listed maven dependencies.

**Reply by laeubi** [reply]
You have different choices here, you can use `bnd-maven-plugin` or the new pde automatic manifest generation with Tycho but need to currently use the copy-dependency goal to first copy your jars in a folder then you can include those in the bundle classpath.

**Reply by git-sources** [reply]
Thanks for your reply. Is there somewhere an example how to get this working? Is it not possible using the target definition and maven dependencies and maybe some custom bnd instructions? At least gertting the jars into the bundle classpath could be done this way, or?

**Reply by laeubi** [reply]
The target platform is only for one jar -> one bundle, technically you can maybe archive this but it might be rather fragile and not very portable.

What was done at m2e though is that we have one [main-bundle](https://github.com/eclipse-m2e/m2e-core/blob/412b8a1a513d07069e6886c6c2f18e39bc41ebee/target-platform/target-platform.target#L128) (here maybe `log4j-api`) and the ones that should be also included are [fragments to the host](https://github.com/eclipse-m2e/m2e-core/blob/412b8a1a513d07069e6886c6c2f18e39bc41ebee/target-platform/target-platform.target#L151) (here `log4j-core`).

So depending on what you want to archive this could be an option for you, but maybe you should also check if these are not already bundles?
